server/text_generation_server/server.py

```diff
@@ -88,16 +98,33 @@ def batch_from_pb(batch):
                 batch, self.model.tokenizer, self.model.dtype, self.model.device
             )
 
-        batches = [batch_from_pb(batch) for batch in request.batches]
-        self.model.warmup(batches)
+        if self.model.batch_type in VLM_BATCH_TYPES :
+            max_supported_total_tokens = self.model.warmup(request)
```

Comment:
> Why don't we get a batch first here? Like in TGI:https://github.com/huggingface/text-generation-inference/blob/v2.0.4/server/text_generation_server/server.py#L110
Because I need use the original request to generate the warmup batches with different sequence length and batch size. Please see the https://github.com/huggingface/tgi-gaudi/pull/187/files#diff-5310651c7992aefadd4915265eaedb0e61e521d4304cc3a0d3fb3e83b0872fdaR1090. I want to modify the truncate and request batch size before calling the batch_from_pb function. By doing this I can reuse batch_from_pb to generate these warmup batches instead of rewriting a function to complete this function after generation.